File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.py
class Solution(object):
    def countNegatives(self, grid):
        count = 0
        for row in grid:
            left = 0
            right = len(row)-1
            while left < right:
                mid = (left + right)//2
                if row[mid] < 0:
                    right = mid
                else:
                    left = mid + 1
            if row[left] < 0:
                count += len(row) - left
        return count          

File: 1351-count-negative-numbers-in-a-sorted-matrix/1351-count-negative-numbers-in-a-sorted-matrix.java
class Solution 
{
    public int countNegatives(int[][] grid) 
    {
        int count = 0;
        for (int r = 0; r < grid.length; r++)
        {
            int left = 0;
            int right = grid[r].length - 1;
            while (left < right)
            {
                int mid = left + (right - left) / 2;
                if (grid[r][mid] < 0)
                    right = mid;
                else
                    left = mid + 1;
            }
            if (grid[r][left] < 0)
                count += grid[r].length - left;
        }
        return count;
    }
}